diffrax/_integrate.py

```diff
@@ -1012,6 +1134,67 @@ def _allocate_output(subsaveat: SubSaveAt) -> SaveState:
         lambda s: s.ts * direction, final_state.save_state, is_leaf=is_save_state
     )
     ys = jtu.tree_map(lambda s: s.ys, final_state.save_state, is_leaf=is_save_state)
+
+    # Do root find for exact event times
+    if event is not None:
+        event_mask = final_state.event_mask
+        event_happened = _event_happened(event_mask)
+        tevent = final_state.tprev
```

Comment:
Nit: prefer single-assignment form:
```python
if event.root_finder is None:
    tevent = final_state.tprev
else:
    # all the rest of the branch below
    tevent = ...
```
one of these software best-practices for writing easy-to-read code.